File: Codes from the AI-assistants/[Experimental] binary-trees/Basic Prompt - TBR Mode/BinaryTrees-deepseek-basic-tbr.cpp
#include <iostream>
#include <memory>
#include <cmath>
#include <queue>

struct TreeNode {
    std::unique_ptr<TreeNode> left;
    std::unique_ptr<TreeNode> right;
};

std::unique_ptr<TreeNode> create_balanced_tree(int depth) {
    if (depth <= 0) return nullptr;
    
    auto node = std::make_unique<TreeNode>();
    node->left = create_balanced_tree(depth - 1);
    node->right = create_balanced_tree(depth - 1);
    
    return node;
}

int count_nodes(const std::unique_ptr<TreeNode>& root) {
    if (!root) return 0;
    return 1 + count_nodes(root->left) + count_nodes(root->right);
}

void print_tree_structure(const std::unique_ptr<TreeNode>& root, int level = 0) {
    if (!root) return;
    
    std::cout << std::string(level * 2, ' ') << "Node\n";
    print_tree_structure(root->left, level + 1);
    print_tree_structure(root->right, level + 1);
}

int main() {
    int depth = 4;
    auto root = create_balanced_tree(depth);
    
    std::cout << "Created balanced tree with depth " << depth << std::endl;
    std::cout << "Total nodes: " << count_nodes(root) << std::endl;
    std::cout << "Expected nodes: " << (int)std::pow(2, depth) - 1 << std::endl;
    
    std::cout << "\nTree structure:\n";
    print_tree_structure(root);
    
    return 0;
}

File: Codes from the AI-assistants/[Experimental] binary-trees/Basic Prompt - TBR Mode/BinaryTrees-deepseek-basic-tbr.java
import java.util.LinkedList;
import java.util.Queue;

class TreeNode {
    TreeNode left;
    TreeNode right;
}

public class BalancedTree {
    
    public static TreeNode createBalancedTree(int depth) {
        if (depth <= 0) return null;
        
        TreeNode node = new TreeNode();
        node.left = createBalancedTree(depth - 1);
        node.right = createBalancedTree(depth - 1);
        
        return node;
    }
    
    public static TreeNode createBalancedTreeIterative(int depth) {
        if (depth <= 0) return null;
        
        TreeNode root = new TreeNode();
        Queue<TreeNode> currentLevel = new LinkedList<>();
        currentLevel.offer(root);
        
        for (int i = 1; i < depth; i++) {
            Queue<TreeNode> nextLevel = new LinkedList<>();
            while (!currentLevel.isEmpty()) {
                TreeNode node = currentLevel.poll();
                node.left = new TreeNode();
                node.right = new TreeNode();
                nextLevel.offer(node.left);
                nextLevel.offer(node.right);
            }
            currentLevel = nextLevel;
        }
        
        return root;
    }
    
    public static int countNodes(TreeNode root) {
        if (root == null) return 0;
        return 1 + countNodes(root.left) + countNodes(root.right);
    }
    
    public static void printTreeStructure(TreeNode root, int level) {
        if (root == null) return;
        
        System.out.println("  ".repeat(level) + "Node");
        printTreeStructure(root.left, level + 1);
        printTreeStructure(root.right, level + 1);
    }
    
    public static void main(String[] args) {
        int depth = 4;
        TreeNode root = createBalancedTree(depth);
        
        System.out.println("Created balanced tree with depth " + depth);
        System.out.println("Total nodes: " + countNodes(root));
        System.out.println("Expected nodes: " + (Math.pow(2, depth) - 1));
        
        System.out.println("\nTree structure:");
        printTreeStructure(root, 0);
    }
}